iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/KubernetesChecksVisitorTest.java

```diff
@@ -174,6 +174,18 @@ void shouldReportWithSecondaryLocationOnDifferentFile() {
     verify(inputFileContext, times(1)).reportIssue(RuleKey.of("testRepo", "testRule"), range, "testIssue", List.of(secondaryLocation));
   }
 
+  @Test
+  void shouldDiscardSecondaryLocationOnDifferentFileWhenDisabled() {
+    var inputFileContext = createInputFileContextMock(false);
+    when(inputFileContext.sensorContext.config().getBoolean(KubernetesChecksVisitor.DISABLE_SECONDARY_LOCATIONS_IN_OTHER_YAML_KEY + ".testRule")).thenReturn(Optional.of(true));
+    TextRange range = range(1, 0, 1, 1);
+    var secondaryLocationSameFile = new SecondaryLocation(range(1, 0, 2, 3), "testIssueSecondary");
+    var secondaryLocationOtherFile = new SecondaryLocation(range(1, 0, 2, 3), "testIssueSecondary", "my/other/file.yaml");
+    var customVisitor = prepareVisitorToRaise("testIssue", List.of(secondaryLocationSameFile, secondaryLocationOtherFile));
+    customVisitor.scan(inputFileContext, tree);
+    verify(inputFileContext, times(1)).reportIssue(RuleKey.of("testRepo", "testRule"), range, "testIssue", List.of(secondaryLocationSameFile));
```

Comment:
Nitpik. The `verify(inputFileContext)` is an equivalent of `verify(inputFileContext, times(1))`. 
I already fixed it in my PR https://github.com/SonarSource/sonar-iac/pull/1411 , so please leave it for now.